### PR TITLE
Add ability to override private key file path in config and cli

### LINKED ISSUE #50: Add config property for custom priv key share path
It can be desirable to have the priv key share at a different path than `share.json`, e.g. a different mounted filesystem.

### PR DESCRIPTION
**Pull request by agouin**
Adds a `config.yaml` string property `key_file`.

Adds a `--keyfile` or `-k` flag to `horcrux config init` to be able to set a custom file path for the key/key-share file when generating the `config.yaml`. Documentation for this flag added to migration doc.

Removes confusing toml properties on the internal Config struct.

Fixes typo in systemd example.

Resolves #50

### CODE REVIEW

File: cmd/horcrux/cmd/cosigner.go [hunk 1]
```diff
@@ -44,9 +44,16 @@ func StartCosignerCmd() *cobra.Command {
 				cfg      signer.Config
 			)
 
+			var privValKeyFile string
+			if config.PrivValKeyFile == "" {
```
Comment (jackzampolin) [inline cmd/horcrux/cmd/cosigner.go:48]: kk this is fine

File: cmd/horcrux/cmd/config.go [hunk 2]
```diff
@@ -511,6 +518,7 @@ func setChainIDCmd() *cobra.Command {
 
 type Config struct {
 	HomeDir        string          `json:"home-dir" yaml:"home-dir"`
+	PrivValKeyFile string          `json:"key_file" yaml:"key_file"`
```
Comment (jackzampolin) [inline cmd/horcrux/cmd/config.go:521]: maybe add `omitempty` here?

File: signer/Config.go [hunk 3]
```diff
@@ -3,44 +3,28 @@ package signer
...
-	PrivValStateDir   string           `toml:"state_dir"`
-	ChainID           string           `toml:"chain_id"`
-	CosignerThreshold int              `toml:"cosigner_threshold"`
-	ListenAddress     string           `toml:"cosigner_listen_address"`
-	RaftListenAddress string           `toml:"cosigner_raft_listen_address"`
-	Nodes             []NodeConfig     `toml:"node"`
-	Cosigners         []CosignerConfig `toml:"cosigner"`
-}
-
-func LoadConfigFromFile(file string) (Config, error) {
```
Comment (jackzampolin) [inline signer/Config.go:32]: is this no longer used?

Comment (agouin) [reply signer/Config.go:32]: no usages in the project. wasn't a recent change from what I can tell, just dead code that's been in here a while.

File: cmd/horcrux/cmd/config.go (hunk 2 above)
Comment (agouin) [reply cmd/horcrux/cmd/config.go]: done

**Review by jackzampolin — APPROVED** [review]

**Pull request merged**